File: binary_search/binary_search.ts
const binary_search = ( arr: number[], value: number ): number => {
    let low: number = 0;
    let high: number = arr.length - 1;

    while( low <= high ) {
        let mid: number = Math.trunc((low + high) / 2);

        if( arr[mid] == value ) {
            return mid;
        } else if( arr[mid] > value ) {
            high = mid - 1;
        } else {
            low = mid + 1;
        }
    }

    return -1;
}

File: binary_search/python/binary_search.py
def binary_search(arr: list[int], value: int) -> int:
    start: int = 0
    end: int = len(arr) - 1

    while start <= end:
        mid: int = (start + end) // 2

        if arr[mid] == value:
            return mid
        elif arr[mid] > value:
            end = mid - 1
        else:
            start = mid + 1
    
    return -1

if __name__ == '__main__':
    arr: list[int] = [2, 5, 8, 12, 16, 23, 38, 56, 72, 91]
    val: int = 23

    search: int = binary_search(arr, val)

    if search != -1:
        print(f'Valor {val} en index: {search}')
    else:
        print(f'Valor {val} no encontrado en la lista')